thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -274,6 +274,34 @@ class ThinReplicaImpl {
       return subscribe_status;
     }
 
+    // If last_known + 1 update requested keep waiting until we have at least one live update
+    // Note that TRS considers last_known + 1 update request as valid, even if the requested
+    // update doesn't exist in storage yet.
+    bool next_update_available = false;
+    while (not next_update_available) {
+      if (request->has_events()) {
+        // Determine latest block available
+        auto last_block_id = (config_->rostorage)->getLastBlockId();
```

Comment:
For every iteration of the while-loop, we are asking storage for the last block id. Don't we only care about the last block id once? If the last block id changes then we are good to go and so we wouldn't even need the live update queue.

Reply:
I agree, updated.